test/utf8_test.cpp

```diff
@@ -0,0 +1,38 @@
+
+#include <gtest/gtest.h>
+
+#include "utils/utf8.hpp"
+
+namespace devilution {
+namespace {
+
+TEST(AppendUtf8Test, OneByteCodePoint)
+{
+	std::string s;
+	AppendUtf8(U'a', s);
+	EXPECT_EQ(s, "a");

```

Comment:
Despite the function being called AppendUtf8 all current uses are effectively initialisation. If we decided to change the internals to assign instead of append this would be a significant behaviour difference (despite having no impact on the current code). I would expect at least one test to break.

```cpp
s = "test";
AppendUtf8(U'a', s);
EXPECT_EQ(s, "testa");
```

Reply:
Done